### DISCUSSION TITLE
S3-WIFI-ST no longer communicates with Solis Cloud

### DISCUSSION THREAD

**Discussion by Samikouk** [alienatedsec/solis-ha-modbus-cloud#37] [General]
@alienatedsec
A great repo and great work, I have a Solis Inverter with S3-WIFI-ST datalogger, with that setup I managed to get data to my HA, but as soon as I plug the datalogger to the rest of the system, no data flows to Solis Cloud. I can see that the TCP Client is responsible for that communication, but not sure how that is being routed out, or am I missing anything here? Thanks in advance

**Comment by alienatedsec** [comment]
@Samikouk Which IP you are using for the HA integration?

**Reply by Samikouk** [reply]
The TCP server IP

**Reply by alienatedsec** [reply]
@Samikouk On `TCP Client` config, you need to point to the TCP Server in this section

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/7dedd1b7-66c7-4f22-a611-c6ce33da8c83)

**Reply by Samikouk** [reply]
@alienatedsec thank you for your response, please see below my current configuration.

TCP Server
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/20213848/c051b1fd-f65b-4fdc-8848-9f0545c62857)

TCP Client
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/20213848/36bdd98b-1e52-4344-a870-b6db2f4b5a9c)

So the TCP Client is pointing at the TCP Server, I am getting all metrics to HA but as soon as I plug the datalogger to this configuration, it stops talking to Solis Cloud. 
Which component should be perfomring this step? I probably need to run wireshark on my machine to trace the root issue.

**Reply by alienatedsec** [reply]
@Samikouk What is in `more advanced settings` for the `TCP Client`?

**Reply by Samikouk** [reply]
@alienatedsec 
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/20213848/69943199-223c-4213-a475-699fba06166e)

**Reply by alienatedsec** [reply]
@Samikouk Imagine that your `TCP Client` is the actual inverter in this scenario, and the datalogger `S3-WIFI-ST` (not `TCP Client`) sends/passes the information from the inverter to the SolisCloud.
